js/src/traceable.ts

```diff
@@ -29,43 +29,64 @@ AsyncLocalStorageProviderSingleton.initializeGlobalInstance(
   new AsyncLocalStorage<RunTree | undefined>()
 );
 
-const handleRunInputs = (rawInputs: unknown[]): KVMap => {
+const handleRunInputs = (
+  rawInputs: unknown[],
+  processInputs: (inputs: Readonly<KVMap>) => KVMap
+): KVMap => {
   const firstInput = rawInputs[0];
+  let inputs: KVMap;
 
   if (firstInput == null) {
-    return {};
+    inputs = {};
+  } else if (rawInputs.length > 1) {
+    inputs = { args: rawInputs };
+  } else if (isKVMap(firstInput)) {
+    inputs = firstInput;
+  } else {
+    inputs = { input: firstInput };
   }
 
-  if (rawInputs.length > 1) {
-    return { args: rawInputs };
+  try {
+    return processInputs(inputs);
+  } catch (e) {
+    console.error("Error occurred during processInputs:", e);
+    return inputs;
   }
-
-  if (isKVMap(firstInput)) {
-    return firstInput;
-  }
-
-  return { input: firstInput };
 };
 
-const handleRunOutputs = (rawOutputs: unknown): KVMap => {
+const handleRunOutputs = (
+  rawOutputs: unknown,
+  processOutputs: (outputs: Readonly<KVMap>) => KVMap
+): KVMap => {
+  let outputs: KVMap;
+
   if (isKVMap(rawOutputs)) {
-    return rawOutputs;
+    outputs = rawOutputs;
+  } else {
+    outputs = { outputs: rawOutputs };
+  }
+
+  try {
+    return processOutputs(outputs);
+  } catch (e) {
+    console.error("Error occurred during processOutputs:", e);
```

Comment:
_nit_: would also add a explainer that raw outputs / inputs are being sent to LangSmith